### ISSUE TITLE
Test base branch detection

### ISSUE THREAD

**Issue by alexr00** [microsoft/vscode-pull-request-github#4961]
Refs: 

- [ ] anyOS
- [ ] anyOS

Complexity: 3

---

You will need the GitHub Pull Request and Issue pre-release extension with minimum version v0.67.20230627*, which should be available shortly after 11:00 CET on June 27.

1. Set `"githubPullRequests.experimental.createView": true`
2. Check out a branch that you would conceivably want to merge a PR into. For example, the `main` branch, or a `release/*` branch.
3. From that branch, create and check out a topic branch.
4. Make a commit on your topic branch, or not. 
5. Run the "Create Pull Request" command
6. Verify that the "Base" remote/branch are populated with is the branch that you branched off from in step 3, i.e., the branch that you checked out in step 2.
7. Repeat steps 2-6, but start with a different branch in step 2.

**Comment by bpasero** [comment]
This is very very helpful 👏

**Comment by Romainbonte** [comment]
0x438420C38137796BC53E7eB4fa46cf3d65a950cf

**Comment by eleanorjboyd** [comment]
Hi! Works great, I just had one question:

If I am on `main` (which is pushed to remote), then I check out to a new branch `release`, then I check out to new branch `release-1.0` the base of `release-1.0` when I create a PR is `main`. This is the correct behavior right? And only once I go back and publish the branch `release` and then switch to `release-1.0` does it change the base in the PR to `release`.

**Comment by alexr00** [comment]
@eleanorjboyd that is the correct behavior! Did you feel that it was unexpected? My thinking was the following:
- Only branches that are available on the remote make sense to use as PR base.
- When you create branch B from branch A, and branch A doesn't exist on the remote, then we still fall back to the default branch of the repo.

**Comment by eleanorjboyd** [comment]
yeah it made sense to me! Once I saw the first time it was based on `main` I knew exactly what steps I had to do in order to make what I was thinking happen (publish the release branch then get that as a base for my newest branch). Thanks! It is a great feature!
